Mathlib/LinearAlgebra/BilinearForm/Basic.lean

```diff
@@ -46,236 +47,183 @@ Bilinear form,
 
 open BigOperators
 
-universe u v w
+open LinearMap (BilinForm)
 
-/-- `BilinForm R M` is the type of `R`-bilinear functions `M → M → R`. -/
-structure BilinForm (R : Type*) (M : Type*) [CommSemiring R] [AddCommMonoid M] [Module R M] where
-  bilin : M → M → R
-  bilin_add_left : ∀ x y z : M, bilin (x + y) z = bilin x z + bilin y z
-  bilin_smul_left : ∀ (a : R) (x y : M), bilin (a • x) y = a * bilin x y
-  bilin_add_right : ∀ x y z : M, bilin x (y + z) = bilin x y + bilin x z
-  bilin_smul_right : ∀ (a : R) (x y : M), bilin x (a • y) = a * bilin x y
-#align bilin_form BilinForm
+universe u v w
 
 variable {R : Type*} {M : Type*} [CommSemiring R] [AddCommMonoid M] [Module R M]
 variable {S : Type*} [CommSemiring S] [Algebra S R] [Module S M] [IsScalarTower S R M]
 variable {R₁ : Type*} {M₁ : Type*} [CommRing R₁] [AddCommGroup M₁] [Module R₁ M₁]
 variable {V : Type*} {K : Type*} [Field K] [AddCommGroup V] [Module K V]
 variable {B : BilinForm R M} {B₁ : BilinForm R₁ M₁}
 
+namespace LinearMap
+
 namespace BilinForm
 
-instance : CoeFun (BilinForm R M) fun _ => M → M → R :=
-  ⟨bilin⟩
+instance : CoeFun (BilinForm R M) fun _ => M → M → R := ⟨fun B₃ x y => B₃ x y⟩
```

Comment:
I think this instance is dangerous; it applies even when the `BilinForm` alias is not being used, and is what breaks `A x` in the other comment thread in this PR.

Reply:
Even with this instance in place, the lemmas below are pretty useless; you get things like
```
coe_smul : (fun x y ↦ ((a • B) x) y) = a • fun x y ↦ (B x) y
```

Reply:
Okay, I've removed that instance. Thanks.